bootstrap/mod/fabric/src/main/java/org/geysermc/geyser/platform/fabric/GeyserFabricBootstrap.java

```diff
@@ -73,22 +87,96 @@ public void onInitialize() {
+//
+//                        Field positionField = null;
+//
+//                        for (String methodName : positionFields) {
+//                            try {
+//                                positionField = Entity.class.getDeclaredField(methodName);
+//                                positionField.setAccessible(true);
+//                                break;
+//                            } catch (NoSuchFieldException ignored) {}
+//                        }
+//
+//                        if (positionField == null) {
+//                            throw new RuntimeException("Unable to get position from ServerPlayer.");
+//                        }
+//
+//                        List<String> levelMethods = List.of("getLevel", "method_14220", "serverLevel", "method_51469");
+//
+//                        Method levelMethod = null;
+//
+//                        for (String methodName : levelMethods) {
+//                            try {
+//                                levelMethod = player.getClass().getMethod(methodName);
+//                                break;
+//                            } catch (NoSuchMethodException ignored) {}
+//                        }
+//
+//                        if (levelMethod == null) {
+//                            throw new RuntimeException("Unable to get level from ServerPlayer.");
+//                        }
+//
+//                        List<String> serverFields = List.of("server", "field_13995");
+//
+//                        Field serverField = null;
+//
+//                        for (String methodName : serverFields) {
+//                            try {
+//                                serverField = player.getClass().getDeclaredField(methodName);
+//                                serverField.setAccessible(true);
+//                                break;
+//                            } catch (NoSuchFieldException ignored) {}
+//                        }
+//
+//                        if (serverField == null) {
+//                            throw new RuntimeException("Unable to get server from ServerPlayer.");
+//                        }
+//
+//                        // Double casting as in older versions, player is instance of CommandSource, but not in modern versions
+//                        //noinspection RedundantCast
+//                        return new CommandSourceStack((CommandSource) (Object) player, (Vec3) positionField.get(player), player.getRotationVector(), (ServerLevel) levelMethod.invoke(player), player.getPermissionLevel(), player.getName().getString(), player.getDisplayName(), (MinecraftServer) serverField.get(player), player);
+//                    } catch (IllegalAccessException | InvocationTargetException e) {
+//                        throw new RuntimeException(e);
+//                    }
+//                }
+            };
+
+            var sourceConverter = CommandSourceConverter.layered(
```

Comment:
okay after seeing this, this definitely screams mod adapters xd 
I might be able to dedicate some time next week to help out with a basic setup to set them up? Those would also be handy for world manager shenanigans